### PR TITLE
Use default metric registry in Prometheus exporter

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Currently the Prometheus exporter is using a private metrics register. This doesn't work well for BK client where we want to expose in the same registry of application, or when we want to interact with other tools that use the default static registry (eg: log4j, jetty integrations for prometheus).

### CODE REVIEW

File: bookkeeper-stats-providers/prometheus-metrics-provider/src/main/java/org/apache/bookkeeper/stats/prometheus/PrometheusMetricsProvider.java [hunk 1]
```diff
@@ -128,10 +128,17 @@ public void start(Configuration conf) {
...
-        new StandardExports().register(registry);
-        new MemoryPoolsExports().register(registry);
-        new GarbageCollectorExports().register(registry);
-        new ThreadExports().register(registry);
+        try {
+            new StandardExports().register(registry);
+            new MemoryPoolsExports().register(registry);
+            new GarbageCollectorExports().register(registry);
+            new ThreadExports().register(registry);
+        } catch (Exception e) {
```
Comment (sijie) [inline bookkeeper-stats-providers/prometheus-metrics-provider/src/main/java/org/apache/bookkeeper/stats/prometheus/PrometheusMetricsProvider.java]: this approach will cause some exports are not registered. for example, if `StandardExports` was registered but no the others, this approach will cause the others being skipped.

Comment (sijie) [reply bookkeeper-stats-providers/prometheus-metrics-provider/src/main/java/org/apache/bookkeeper/stats/prometheus/PrometheusMetricsProvider.java]: I would suggest having a common method that catches exception. 

```
public void registerExports(Exports exports, CollectorRegistry registry) {
     try {
            exports.register(registry);
     } catch (Exception e) {
            ....
     }

}
```

Comment (merlimat) [reply bookkeeper-stats-providers/prometheus-metrics-provider/src/main/java/org/apache/bookkeeper/stats/prometheus/PrometheusMetricsProvider.java]: @sijie Fixed

**Comment by merlimat** [comment]
run java9 tests

**Comment by merlimat** [comment]
rebuild java9

**Comment by sijie** [comment]
run integration tests

**Comment by merlimat** [comment]
run integration tests

**Comment by eolivelli** [comment]
@sijie. Do you want this on branch 4.8?

**Comment by sijie** [comment]
@eolivelli yes. I was in the process of merging it. :)

**Pull request merged**